src/components/Form/Form.tsx

```diff
@@ -0,0 +1,91 @@
+/* eslint-disable jsx-a11y/control-has-associated-label */
+
+import React, {
+  useEffect,
+  useMemo,
+  useRef,
+  useState,
+} from 'react';
+import classNames from 'classnames';
+import { Todo } from '../../types/Todo';
+import { USER_ID } from '../../constants/USER_ID';
+
+type Props = {
+  loading: boolean,
+  todos: Todo[]
+  addTodo: (newTodo: Todo) => void
+  reverteCompletedTodo: (todosForRevert: Todo[]) => void,
+  titleFocus: boolean,
+};
+
+export const Form: React.FC<Props> = React.memo(
+  ({
+    loading,
+    todos,
+    addTodo,
+    reverteCompletedTodo,
+    titleFocus,
+  }) => {
+    const [value, setValue] = useState('');
+    const formInput = useRef<HTMLInputElement | null>(null);
+    const [isFirstRender, setIsFirstRender] = useState(true);
+
+    const activeTodos = useMemo(() => {
+      return todos.filter(todo => !todo.completed);
+    }, [todos]);
+
+    useEffect(() => {
+      if (isFirstRender) {
+        setIsFirstRender(false);
+
+        return;
+      }
+
+      if (formInput.current) {
+        formInput.current.focus();
+      }
+    }, [titleFocus]);
+
+    const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+      event.preventDefault();
+
+      const newTodo = {
+        id: 0,
```

Comment:
I don't think `id: 0` is good for reach mapping 😄

Reply:
> I don't think `id: 0` is good for reach mapping 😄

fixed

Reply:
Is it? I still can see it in the code ;P